packages/dapp/components/redeem/UcrRedeem.tsx

```diff
@@ -54,12 +58,49 @@ const UcrRedeem = () => {
 
   const submitEnabled = !!(extractValidAmount() && !doingTransaction);
 
+  const handleMax = () => {
+    const ucrValue = ethers.utils.formatEther(balances.ucr);
+    setInputVal(parseInt(ucrValue).toString());
+  };
+
+  function onChangeValue(e: React.ChangeEvent<HTMLInputElement>) {
+    setSelectedRedeemToken(e.target.value);
+  }
+
   return (
     <div>
-      <PositiveNumberInput placeholder="uCR Amount" value={inputVal} onChange={setInputVal} />
+      <div onChange={onChangeValue}>
+        <p>Please select a token to redeem for:</p>
+        <input type="radio" id="tokenChoice1" name="redeemToken" value="uAD" checked={selectedRedeemToken === "uAD"} readOnly />
+        <label htmlFor="tokenChoice1">uAD</label>
+
+        <input type="radio" id="tokenChoice2" name="redeemToken" value="USDC" checked={selectedRedeemToken === "USDC"} readOnly />
+        <label htmlFor="tokenChoice2">USDC</label>
+
+        <input type="radio" id="tokenChoice3" name="redeemToken" value="DAI" checked={selectedRedeemToken === "DAI"} readOnly />
+        <label htmlFor="tokenChoice3">DAI</label>
+
+        <input type="radio" id="tokenChoice4" name="redeemToken" value="USDT" checked={selectedRedeemToken === "USDT"} readOnly />
+        <label htmlFor="tokenChoice4">USDT</label>
+      </div>
+      <div>
+        <PositiveNumberInput placeholder="uCR Amount" value={inputVal} onChange={setInputVal} />
+        <span onClick={handleMax}>MAX</span>
+      </div>
+      {inputVal && quotePrice && (
+        <div>
+          {inputVal} uCR -&gt; {quotePrice} {selectedRedeemToken}.
+        </div>
+      )}
       <Button onClick={handleRedeem} disabled={!submitEnabled}>
-        Redeem uCR for uAD
+        Redeem uCR for {selectedRedeemToken}
       </Button>
+      <div>----------------------------------------------------------------------------</div>
+      <div>Uniswap SwapWidget (customizable)</div>
+      <div>----------------------------------------------------------------------------</div>
```

Comment:
These artifacts are fine to leave in as long as we merge to an interim (new branch) between your PR and `development` because I plan to personally go in to work on the CSS/aesthetic details.